apple/DemoApp/Demo/DemoModel.swift

```diff
@@ -133,29 +133,26 @@ let demoModel = DemoModel()
         return .idle
     }
 
-    private func wrap(wrap: () throws -> DemoAppState) {
-        do {
-            errorMessage = nil
-            appState = try wrap()
-        } catch {
-            errorMessage = error.localizedDescription
-            appState = .idle
-        }
-    }
-
     private func wrap(wrap: () async throws -> DemoAppState) async {
         do {
-            errorMessage = nil
-            appState = try await wrap()
+            await MainActor.run {
+                errorMessage = nil
+            }
+            let newState = try await wrap()
+            await MainActor.run {
+                appState = newState
+            }
         } catch {
-            errorMessage = error.localizedDescription
-            appState = .idle
+            await MainActor.run {
+                errorMessage = error.localizedDescription
+                appState = .idle
+            }
         }
     }
 
-    func chooseDestination() {
-        wrap {
-            try appState.setDestination(destination)
+    func chooseDestination() async {
+        await wrap {
+            try await appState.setDestination(destination)
```

Comment:
This doesn't actually do anything async, so that triggers a compiler warning FWIW.